### PR TITLE
Snapshot

### LINKED ISSUE #276: Jest snapshots
### What's up?

Using [Jest snapshots](https://facebook.github.io/jest/blog/2016/07/27/jest-14.html), we can turn Cosmos into an automatic testing suite on top of existing fixtures.

### Mkay, tell me more...

#268 and #272 lead up to this. Now we need to do 3 things:

- [x] Allow loader to be inited with `component` and `fixture` already selected (before it always initialised in a blank state and waited for an onMessage event from parent frame) 
- [x] Expose a `createLoaderElement` fn from react-cosmos package
- [x] Create new package `react-cosmos-telescope` (?) that 
  a. loads the config from react-cosmos-config
  b. gets paths from react-cosmos-voyager
  c. imports and iterates thru all paths, creating a Jest test for each

### PR DESCRIPTION
**Pull request by skidding**
Fixes #276

### CODE REVIEW

File: packages/react-cosmos-telescope/src/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const proxies = cosmosConfig.proxies.map(proxy => require(proxy));
+  const components = importFileTree(filePaths.components);
+  const fixtures = keys(filePaths.fixtures).reduce((acc, component) => {
+    return {
+      ...acc,
+      [component]: importFileTree(filePaths.fixtures[component]),
+    };
+  }, {});
+
+  Object.keys(fixtures).forEach(component => {
```
Comment (ovidiubute) [inline packages/react-cosmos-telescope/src/index.js]: `keys(fixtures).forEach(component => {`

File: packages/react-cosmos-telescope/src/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const componentFixtures = fixtures[component];
+    Object.keys(componentFixtures).forEach(fixture => {
```
Comment (ovidiubute) [inline packages/react-cosmos-telescope/src/index.js]: `keys(componentFixtures).forEach(fixture => {`

File: packages/react-cosmos/src/__tests__/create-loader-element.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  jest.mock('react', () => mockReact);
+  jest.mock('../components/Loader', () => mockLoader);
+  jest.mock('../components/proxies/StateProxy', () => () => mockStateProxy);
+  jest.mock('../components/proxies/PropsProxy', () => mockPropsProxy);
+  jest.mock('../load-modules', () => ({
+    loadComponents: mockLoadComponents,
+    loadFixtures: mockLoadFixtures,
+  }));
+
+  createLoaderElement = require('../create-loader-element.jsx').default;
```
Comment (ovidiubute) [inline packages/react-cosmos/src/__tests__/create-loader-element.js]: Should we still use file extensions in `require`? Maybe we can set the trend for removing it for good, I don't see the value in it at all.

Comment (skidding) [inline packages/react-cosmos/src/__tests__/create-loader-element.js]: I'm with you, already battled with ESLint and [managed to remove most of them here](https://github.com/react-cosmos/react-cosmos/pull/271/commits/adf592102da5f8b88f13008a355e73fbf7f935be)

File: packages/react-cosmos/src/start-loader.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import reactDomPolyfill from 'react-dom-polyfill';
+
+const ReactDOM = reactDomPolyfill(React);
+
+let domContainer;
+
+const createDomContainer = () => {
+  if (!domContainer) {
+    domContainer = document.createElement('div');
+    domContainer.style.position = 'absolute';
```
Comment (RadValentin) [inline packages/react-cosmos/src/start-loader.js]: Maybe assign these?

```js
Object.assign(domContainer.style, {
  position: 'absolute',
  width: '100%',
  height: '100%',
  background: 'pink'
});
```

**Comment by skidding** [comment]
Published under `2.0.0-beta.7` – get it while it's hot! 🥖  🍞

**Pull request merged**